server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1189,6 +1208,26 @@ private static class MetadataDiff implements Diff<Metadata> {
             new DiffableUtils.DiffableValueReader<>(ReservedStateMetadata::readFrom, ReservedStateMetadata::readDiffFrom);
 
         MetadataDiff(StreamInput in) throws IOException {
+            if (in.getVersion().onOrAfter(NOOP_METADATA_DIFF_VERSION)) {
+                noop = in.readBoolean();
+                if (noop) {
+                    // noop diff, other fields are irrelevant
+                    clusterUUID = null;
+                    clusterUUIDCommitted = false;
+                    version = -1L;
+                    coordinationMetadata = null;
+                    transientSettings = null;
+                    persistentSettings = null;
+                    hashesOfConsistentSettings = DiffableUtils.emptyDiff();
+                    indices = DiffableUtils.emptyDiff();
+                    templates = DiffableUtils.emptyDiff();
+                    customs = DiffableUtils.emptyDiff();
+                    reservedStateMetadata = DiffableUtils.emptyDiff();
```

Comment:
I wonder if it is possible or worth replacing constructor with a static factory? This way we could have a constant static empty diff

Reply:
Good point, we don't need a specific instance when reading. We even have a static factory already. I made use of that and reused the existing empty instance in `SimpleDiffable` now + made the constructor private to avoid abuse.